lambda_src/drivers/process_https.py

```diff
@@ -71,130 +83,136 @@ def process_row(
-                        ).digest()
-                    ).decode()
-                ),
-            },
-        )
-
-        req_auth = (
-            loads(auth)
-            if auth and auth.startswith('{')
-            else parse_header_dict(auth)
-            if auth
-            else {}
-        )
-        auth_host = req_auth.get('host')
-
-        # We reject the request if the 'auth' is present but doesn't match the pinned host.
-        if auth_host and req_host and auth_host != req_host:
-            raise ValueError(
-                "Requests can only be made to host provided in the auth header."
-            )
-        # If the URL is missing a hostname, use the host from the auth dictionary
-        elif auth_host and not req_host:
-            req_host = auth_host
-        # We make unauthenticated request if the 'host' key is missing.
-        elif not auth_host:
-            raise ValueError(f"'auth' missing the 'host' key.")
-        elif 'basic' in req_auth:
-            req_headers['Authorization'] = make_basic_header(req_auth['basic'])
-        elif 'bearer' in req_auth:
-            req_headers['Authorization'] = f"Bearer {req_auth['bearer']}"
-        elif 'authorization' in req_auth:
-            req_headers['authorization'] = req_auth['authorization']
-        elif 'headers' in req_auth:
-            req_headers.update(req_auth['headers'])
-        elif 'body' in req_auth:
-            if json:
-                raise ValueError(f"auth 'body' key and json param are both present")
-            if data:
-                raise ValueError(f"auth 'body' key and data param are both present")
-            else:
-                data = (
-                    req_auth['body']
-                    if isinstance(req_auth['body'], str)
-                    else dumps(req_auth['body'])
-                )
 
     # query, nextpage_path, results_path
     req_results_path: str = results_path
     req_cursor: str = cursor
+    req_page_count: int = 0
     req_method: str = method.upper()
+    req_data: Optional[bytes] = None if data is None else data.encode()
     if json:
-        req_data: Optional[bytes] = (
-            json if json.startswith('{') else dumps(parse_header_dict(json))
-        ).encode()
+        req_json = loads(json) if json.startswith('{') else parse_header_dict(json)
         req_headers['Content-Type'] = 'application/json'
     else:
-        req_data = None if data is None else data.encode()
+        req_json = None
 
     next_url: Optional[str] = req_url
     row_data: List[Any] = []
+    metadata: Optional[Any] = None
 
     LOG.debug('Starting pagination.')
     while next_url:
-        LOG.debug(f'next_url is {next_url}.')
-        req = Request(next_url, method=req_method, headers=req_headers, data=req_data)
+        parsed_url = urlparse(next_url)
+
+        # We look for an auth header and if found, we parse it from its encoded format
+        if auth:
+            auth = render_jinja_template(
```

Comment:
to debug this started by looking through the free variables in the expression and blocks to abstract those relationships as a function.

in this specific expression,`(auth, parsed_url, method, ...)` is used as input and then overwrites `auth`. this is fine if it's run once, but inside of a loop, we need to preserve the original `auth` that holds the template. it's often questionable practice to re-use a variable name like this and this is a decent example of why.